v2/context/context.go

```diff
@@ -50,3 +51,29 @@ func TopicFrom(ctx context.Context) string {
 	}
 	return ""
 }
+
+// Opaque key type used to store linear backoff parameters
+type linearBackoffKeyType struct{}
+
+var linearBackoffKey = linearBackoffKeyType{}
+
+// WithLinearBackoff returns back a new context with the linear backoff parameters.
+// MaxTries is the maximum number for retries and delay is the time interval between retries
+func WithLinearBackoff(ctx context.Context, delay time.Duration, maxTries int) context.Context {
+	return context.WithValue(ctx, linearBackoffKey, Backoff{
+		Strategy: BackoffStrategyLinear,
+		Period:   delay,
+		MaxTries: maxTries,
+	})
+}
+
+// RetryFrom looks in the given context and returns the linear backoff parameters if found, otherwise 0,0
+func LinearBackoffFrom(ctx context.Context) (time.Duration, int) {
```

Comment:
I would have this return a `Backoff` struct.

Reply:
Also renamed to `BackoffFrom`

Reply:
`Backoff` seems to be confusing. I would rename to `Retry` or `Retires`

Reply:
I updated the comment but not the code. Distracted me. I'll go with Retries

Reply:
fixed.